lib/eth/abi/type.rb

```diff
@@ -138,6 +161,7 @@ def validate_base_type(base_type, sub_type)
 
           # bytes can be no longer than 32 bytes
           raise ParseError, "Maximum 32 bytes for fixed-length string or bytes" unless sub_type.empty? || sub_type.to_i <= 32
+        when "tuple"
```

Comment:
any idea how to validate this?

Reply:
what about:

```ruby
raise ParseError if (sub_type.dimensions.size != 1 || sub_type.dimensions.first&.zero?) && !sub_type.is_dynamic?
```

Just taking a stab at it